client/image_create.go

```diff
@@ -26,15 +28,47 @@ func (cli *Client) ImageCreate(ctx context.Context, parentReference string, opti
 	if options.Platform != "" {
 		query.Set("platform", strings.ToLower(options.Platform))
 	}
-	resp, err := cli.tryImageCreate(ctx, query, options.RegistryAuth)
+	resp, err := cli.tryImageCreate(ctx, query, staticAuth(options.RegistryAuth))
 	if err != nil {
 		return nil, err
 	}
 	return resp.Body, nil
 }
 
-func (cli *Client) tryImageCreate(ctx context.Context, query url.Values, registryAuth string) (*http.Response, error) {
-	return cli.post(ctx, "/images/create", query, nil, http.Header{
-		registry.AuthHeader: {registryAuth},
-	})
+func (cli *Client) tryImageCreate(ctx context.Context, query url.Values, resolveAuth registry.RequestAuthConfig) (*http.Response, error) {
+	hdr := http.Header{}
+	var lastErr error
+	for {
+		registryAuth, tryNext, err := getAuth(ctx, resolveAuth)
+		if err != nil {
+			// TODO(thaJeztah): should we return an "unauthorised error" here to allow the caller to try other options?
+			if errors.Is(err, errNoMorePrivilegeFuncs) && lastErr != nil {
+				return nil, lastErr
+			}
+			return nil, err
+		}
+		if registryAuth != "" {
+			hdr.Set(registry.AuthHeader, registryAuth)
+		}
+		resp, err := cli.post(ctx, "/images/create", query, nil, hdr)
+		if err == nil {
+			// Discard previous errors
+			return resp, nil
+		} else {
```

Comment:
This won't work when pulling an image from a private repository. It produces a 404 instead of a 403/401 which would then never call the next PrivilegeFunc.